typings/temp.d.ts

```diff
@@ -0,0 +1,4 @@
+declare module 'temp' {
+    function track(): void;
+    function mkdir(prefix: string): (err: Error, dirPath: string) => void;
```

Comment:
`mkdir` *returns* a callback? Seems unlikely - this should be the 2nd argument, and mkdir probably returns void.

This suggests the type you're getting when you `promisify` this isn't right either atm. Looks like the only reason you don't hit an implicit any is because the only place you call it you then immediately specify the next promise argument type.

Promisify does return the correct promisified type if the function you pass to promisify takes a callback as the last argument, but if it doesn't then it'll return a plain `Function` (i.e. callable, but takes and returns `any`). The types are little hacky but kind of interesting, if you're into that sort of thing: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/bluebird/index.d.ts#L390-L403. It means it works most of the time, but it's not totally foolproof. Implicit any should catch any substantial issues generally though.